masonry_core/src/app/render_root.rs

```diff
@@ -619,6 +711,49 @@ impl RenderRoot {
         res
     }
 
+    /// Add a new layer at the end of the stack, with the given widget as its root, at the given position.
+    pub fn add_layer(&mut self, root: NewWidget<impl Widget + ?Sized>, pos: Point) {
+        tracing::debug!("added layer to stack");
+        mutate_widget(self, self.root_id(), |mut layer_stack| {
+            let mut layer_stack = layer_stack.downcast::<LayerStack>();
+            LayerStack::add_layer(&mut layer_stack, root, pos);
+        });
+
+        self.run_rewrite_passes();
+    }
+
+    /// Removes the layer with the given widget as root.
+    ///
+    /// The base layer cannot be removed.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the intended layer is the base layer in debug mode.
+    pub fn remove_layer(&mut self, root_id: WidgetId) {
+        mutate_widget(self, self.root_id(), |mut layer_stack| {
+            let mut layer_stack = layer_stack.downcast::<LayerStack>();
+            LayerStack::remove_layer(&mut layer_stack, root_id);
+        });
+
+        self.run_rewrite_passes();
+    }
```

Comment:
We should document what happens when the id isn't a layer root, whether we make it a panic or a warning.